ci/ci_helper.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python
+
+""" Since github actions are quite difficult to use when writing scriptable things.
+We rely on this script being able to output the unaltered text.
+"""
+
+import os, sys, platform
+
+PLATFORMS_PRIVATE = ('x86_64-ps4', 'arm64-nx64')
+
+DIRNAME=os.path.dirname(os.path.abspath(__file__))
+
+# We use this function to determine if we should skip this platform
+def can_build_private_platform(platform):
+    return False
+
+def repo_name_to_platforms(repository):
+    return None
+
+if os.path.exists(os.path.join(DIRNAME, 'ci_private.py')):
+    import ci_private
+    can_build_private_platform = ci_private.can_build_private_platform
+    repo_name_to_platforms = ci_private.repo_name_to_platforms
+
+def is_platform_private(platform):
+    return platform in PLATFORMS_PRIVATE
+
+def is_platform_supported(platform):
+    if is_platform_private(platform):
+        return can_build_private_platform(platform)
+    return True
+
+def is_platform_supported_by_repo(args):
+    platform=args[0]
+    if not is_platform_private(platform):
+        # By default all public platforms are supported
+        return True
+
+    repository = os.environ.get('GITHUB_REPOSITORY', None)
+    if repository is None:
+        return True # probably a local build
+
+    platforms = repo_name_to_platforms(repository)
+    if platforms is not None and platform in platforms:
+        return True
+
+    return False
+
+def is_repo_private():
+    repository = os.environ.get('GITHUB_REPOSITORY', None)
+    if repository is None:
+        return False # probably a local build
+
+    # get the platforms
+    platforms = repo_name_to_platforms(repository)
+    if platforms is not None:
+        for platform in platforms:
+            if is_platform_private(platform):
+                return True
+    return False
```

Comment:
Perhaps reverse the default strategy?
I.e. "If it's defold/defold" then it's public, otherwise it's private?

Reply:
What about forks of the public repo? Those would fail the check if we look specifically for defold/defold

Reply:
yeah, not sure what the best strategy is yet
I think this is good enough for now

Reply:
Agree